crates/storage/src/schema/revision_0027.rs

```diff
@@ -0,0 +1,19 @@
+/// This migration adds a new table storing compiled CASM classes.
+pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    use anyhow::Context;
+
+    tx.execute_batch(
+        r"-- Stores compiled CASM for Sierra classes.
+        CREATE TABLE contract_code_casm (
```

Comment:
Nit:
```suggestion
        CREATE TABLE class_code_casm (
```
The only reason the original is called `contract_code` is because classes did not exist yet.

Should we not also consider just having a single table instead of two separate?
```
| hash | sierra definition | casm hash | casm definition | 
```

Reply:
Maybe we could rename `contract_code` to `class_code` as well?

Also, I didn't want to merge these tables since for non-Sierra classes there won't be a CASM hash/definition. You can pretty much still get the same via an OUTER JOIN if we need that.

We could do either, I was just preferring the normalized representation.

Reply:
I think its fair to say that eventually the non-Sierra classes will be a huge exception to the rule. But no harm in keeping them separate I suppose.

We can rename the table as well; that would be nice actually.

`casm_code` and `sierra_code`? Or `casm_classes` and `sierra_classes`? Something like that.

Reply:
Something involving `sierra` in its name would be quite misleading: it's _either_ Sierra or a Cairo 0.x class, but _both_ of those currently live in `contract_code`. Maybe something like `class_definition` and `casm_definition`?

Reply:
Oh right. Meh I just dislike that we're mixing casm and sierra definitions..

I think that's partially I suggested a single table to make it more clear. The sierra hash / definition is allowed to be null. However one can probably have different sierra definitions compile to the same casm hash? So one does technically want to normalize that as well.

Eugh. Okay I don't have an immediete better suggestion than `class_definitions` and `casm_definitions`. Although I still don't like it 😛

Reply:
> Oh right. Meh I just dislike that we're mixing casm and sierra definitions..

Technically old Cairo 0.x class definition is _not_ the same as CASM, though.

Reply:
The json structure is different, but the bytecode is the same surely?

Reply:
Yes, the bytecode is the same. Pretty much all other stuff is either missing or represented differently in the JSON...

Reply:
I've pushed a fixup commit with the table rename.